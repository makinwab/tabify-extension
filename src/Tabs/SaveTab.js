import React, { Component } from 'react'
import { Label, Icon, Form, Message, Checkbox } from 'semantic-ui-react'
import { getCurrentTab, removeCurrentTab } from '../Helpers/extensionUtils'
import client, { getEntries, categoryEntries, environment as contentfulClient } from '../Helpers/contentful'
import Tabs from '../Tabs/Tabs'
import App from '../App/App'

const retrieveUser = () => {
  return client.CDAClient.getEntries({ content_type: 'user', 'fields.email': localStorage.user })
}

class SaveTab extends Component {
  constructor (props) {
    super(props)

    this.state = {
      page: 'SaveTab',
      options: [],
      tab: {},
      title: '',
      category: '',
      selected: '',
      note: '',
      createdBy: '',
      loading: false,
      error: null,
      success: null
    }
  }

  componentDidMount () {
    // Clear timeout from App component
    clearTimeout(this.props.messageTimeout)

    // If action is to edit, use already saved tab information
    // else get the browser's current tab
    if (this.props.editableTab) {
      getEntries({ content_type: 'tab', 'sys.id': this.props.editableTab })
        .then(result => {
          const tab = result.items[0].fields

          this.setState({
            tab: {
              url: tab.url
            },
            title: tab.title,
            category: tab.tag.fields.name,
            selected: tab.tag.sys.id,
            note: tab.note
          })
        })
    } else {
      getCurrentTab(tab => {
        this.setState({ tab, title: tab.title || tab.hostname })
      })
    }

    // Get current user and set state accordingly
    retrieveUser().then(result => {
      return this.setState({
        createdBy: result.items[0].sys.id ,
        closeTab: result.items[0].fields.closeTabOnSave
      })
    })

    // Get all categories for the dropdown
    categoryEntries.then(result => {
      let options = result.items.map((value, index) => {
        return {
          key: value.sys.id,
          text: value.fields.name,
          value: value.sys.id
        }
      })

      this.setState({ options })
    })
  }

  handlePageChange = page => ev => {
    ev.preventDefault()
    this.setState({ page: page.next })
  }

  handleFormChange = (ev, { name, value }) => {
    let selected = this.state.selected
    if (name === 'category') {
      selected = value
    }
    this.setState({ [name]: value, selected })
  }

  handleSubmit = ev => {
    ev.preventDefault()

    const { category, tab, title } = this.state

    this.setState({ loading: true })

    if (!title) {
      this.handleFeedbackState({
        message: 'Title is missing',
        details: "Please add a title",
        value: '👇🏽'
      }, 'error')

      return
    }

    if (!category) {
      this.handleFeedbackState({
        message: 'Category is missing',
        details: "Please select a category",
        value: '👇🏽'
      }, 'error')

      return
    }


    this.validateTab(tab.url || tab.href)
    .then(result => {
      if (result) {
        this.handleFeedbackState({
          message: 'Tab already exists',
          details: "You've previously saved this tab",
          value: tab.url || tab.href
        }, 'error')

        return
      }

      retrieveUser().then(result => this.setState({ user: result.items[0].sys.id }))
      this.saveTab()
    })
  }

  handleFeedbackState (feedback, type) {
    this.setState({
      loading: false,
      [type]: feedback
    })

    setTimeout(() => this.setState({[type]: null}), 5000)
  }

  validateTab (tabUrl) {
    return this.props.getFilteredEntries().then(result => {
      if (this.props.editableTab) {
        return false
      }
      return result.some(value => {
        return value.fields.url === tabUrl
      })
    })
  }

  saveTab ({ tab, note, selected, createdBy, title } = this.state) {
    const payload = {
      title: {
        'en-US': title
      },
      note: {
        'en-US': note
      },
      url: {
        'en-US': tab.url || tab.href
      },
      tag: {
        'en-US': {
          sys: {
            type: 'Link',
            linkType: 'Entry',
            id: selected
          }
        }
      },
      createdBy: {
        'en-US': {
          sys: {
            type: 'Link',
            linkType: 'Entry',
            id: createdBy
          }
        }
      }
    }

    const tabToEdit = this.props.editableTab
    let result

    if (tabToEdit) {
      result = this.updateTab(payload, tabToEdit)
    } else {
      result = this.createTab(payload)
    }

    result
      .then(() => {
        this.updateOnSaveTabStatus(this.state)
          .then((result) => {
            this.setState({ category: '', note: '' })
            return this.handleFeedbackState(true, 'success')
          })
          .then(() => {
            if (this.state.closeTab && !this.props.editableTab) {
              removeCurrentTab()
            }
          })
      })
      .catch(e => {
        let error = JSON.parse(e.message)

        if (error.status === 500) {
          this.handleFeedbackState({
            message: 'Internal Server Error',
            details: 'Please try this action again later',
            value: ''
          }, 'error')
        } else {
          this.handleFeedbackState({
            message: error.message.message,
            details: error.message.details.errors[0].details,
            value: error.message.details.errors[0].value
          }, 'error')
        }
      })
  }

  createTab (payload) {
    return contentfulClient
      .then(environment => environment.createEntry('tab', { fields: payload }))
      .then(entry => entry.publish())
  }

  updateTab (data, tabId) {
    return contentfulClient
      .then(environment => environment.getEntry(tabId))
      .then(entry => {
        entry.fields = data
        return entry.update()
      })
      .then(entry => entry.publish())
  }

  updateOnSaveTabStatus({ closeTab, user: userId}) {
    const data = {
      email: {
        'en-US': localStorage.user
      },
      closeTabOnSave: {
        'en-US': closeTab
      }
    }
    
    return contentfulClient
      .then(environment => environment.getEntry(userId))
      .then(entry => {
        entry.fields = data
        return entry.update()
      })
      .then(entry => entry.publish())
  }

  toggleCheck = () => {
    this.setState({closeTab: !this.state.closeTab})
  }

  render () {
    const { page, note, category, tab, options, loading, error, success, selected, closeTab, title } = this.state

    return (
      <React.Fragment>
        {(page === 'SaveTab')
          ? <div id='SaveTab' className='save-tab'>
            <div className='page-header'>
              <div className='menu-links'>
                <Label color='black' as='a' onClick={this.handlePageChange({ current: 'SaveTab', next: 'Tabs' })}>
                  <Icon name='linkify' />My Tabs
                </Label>
              </div>

              <Icon className='with-pointer' name='home' size='large' onClick={this.handlePageChange({ current: 'Tabs', next: 'App' })} />
            </div>

            <div className='save-tab-form'>
              {error
                ? <Message
                  error
                  header={error.message}
                  content={error.details + ' - ' + error.value}
                /> : ''
              }
              {success
                ? <Message
                  success
                  header='Tab Saved 🎉'
                  content="You've successfully saved a tab" />
                : ''
              }
              <Message
                attached
                header='Ready to save tab!'
                content='Fill out the fields below for the tab' />

              <Form id='saveTabForm' className='attached fluid segment' onSubmit={this.handleSubmit}>
                <Form.Group widths='equal'>
                  <Form.Input fluid className='disabled-field' label='Tab Title' onChange={this.handleFormChange} placeholder='Enter Tab Title' name='title' value={title} />
                  <Form.Select
                    name='category'
                    onChange={this.handleFormChange}
                    value={selected || category}
                    fluid
                    label='Tab Category'
                    options={options}
                    placeholder='Select Category' />
                </Form.Group>

                <Form.TextArea name='note' value={note} onChange={this.handleFormChange} label='Note' placeholder='Leave a note regarding this tab...' />

                {this.props.editableTab ? '' : 
                  <Form.Field>
                    <Checkbox className='close-tab' toggle label='CLOSE TAB ON SAVE' checked={closeTab} onChange={this.toggleCheck}/>
                  </Form.Field>
                }

                {loading ? <Form.Button loading>Loading</Form.Button> : <Form.Button>Save</Form.Button>}
              </Form>

              <Message attached='bottom'>
                <div className='truncate-text'>
                  <Icon name='linkify' />
                  <a className='with-pointer' href={tab.url} target='_blank' rel='noopener noreferrer'>{tab.url}</a>
                </div>
              </Message>
            </div>
          </div>
          : (page === 'App') ? <App />
            : (page === 'Tabs') ? <Tabs getFilteredEntries={this.props.getFilteredEntries} />
              : ''
        }
      </React.Fragment>
    )
  }
}

export default SaveTab
