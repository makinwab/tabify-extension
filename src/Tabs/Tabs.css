/* Save Tab Page */
.save-tab-form {
  color: #ffffff;
  width: 90%;
  margin: 1.5em auto 0 auto;
}

.save-tab-form {
  text-align: center;
  overflow-y: auto;
  overflow-x: hidden;
}

div.disabled-field > div.ui.input {
  opacity: 4 !important;
}

div.truncate-text {
  overflow: hidden;
  text-overflow: ellipsis;
  width: 17em;
  margin: 0 auto;
}

.search .results {
  display: none !important;
}

.no-result {
  color: white;
  text-align: center;
}

/* Tabs Page */
.tabs-header {
  position: relative;
  height: 5em;
}

.tabs-header .home {
  position: absolute;
  left: 0.5em;
  top: 1em;
  color: #F9D873;
}

.tabs-tag {
 color: #F9D873;
}

.tabs-main {
  width:90%;
  margin: 0.5em auto 0 auto;
}

.tabs-main .search-box {
  margin-bottom: 3em;
}

.tabs-main .note, .tabs-main .list-icon {
  color: white !important;
  line-height: 2em;
}

.tabs-main .tab-meta {
 color: grey !important;
 font-style: italic;
}

.with-pointer {
  cursor: pointer;
}