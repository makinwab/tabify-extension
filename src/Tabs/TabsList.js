import React from 'react'
import { List, Divider } from 'semantic-ui-react'

const TabsList = (props) => (
  <React.Fragment>
    <List.Item>
      <List.Icon className='list-icon' name='linkify' size='large' verticalAlign='middle' />

      <List.Content>
        <List.Header>
          <a href={props.fields.url} target='_blank' rel='noopener noreferrer'>{props.fields.title}</a>
          <span className='tabs-tag'> ( #{props.fields.tag.fields.name} )</span>
        </List.Header>
        <List.Description className='note'>{props.fields.note}</List.Description>
        <List.Description className='tab-meta'>Saved at {props.createdAt}</List.Description>
      </List.Content>
    </List.Item>

    <Divider />
  </React.Fragment>
)

export default TabsList
