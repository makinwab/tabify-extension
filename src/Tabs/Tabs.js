import React, { Component } from 'react'
import { Search, List, Label, Icon } from 'semantic-ui-react'
import TabsList from './TabsList'
import SaveTab from './SaveTab'
import App from '../App/App'

import './Tabs.css'

class Tabs extends Component {
  constructor (props) {
    super(props)

    this.state = {
      entries: [],
      page: 'Tabs'
    }

    this.handlePageChange = this.handlePageChange.bind(this)
  }

  componentDidMount () {
    this.invokeTabEntries()
  }

  invokeTabEntries () {
    this.props.getFilteredEntries().then(result => {
      this.setState({ entries: result })
    })
  }

  handlePageChange (ev, page) {
    ev.preventDefault()
    this.setState({ page: page.next })
  }

  render () {
    const { page } = this.state
    return (
      <React.Fragment>
        {(page === 'Tabs') ? <div id='Tabs' className='Tabs'>
          <div className='tabs-header'>
            <div className='menu-links'>
              <Label color='black' as='a' className='with-pointer' onClick={ev => this.handlePageChange(ev, { current: 'Tabs', next: 'SaveTab' })}>
                <Icon name='add' /> Save Tab
              </Label>
            </div>

            <Icon className='with-pointer' name='home' size='large' onClick={ev => this.handlePageChange(ev, { current: 'Tabs', next: 'App' })} />
          </div>

          <div className='tabs-main'>
            <center className='search-box'>
              <Search size='big' />
            </center>
            <List relaxed>
              {this.state.entries.map(value => {
                return (<TabsList key={value.sys.id} createdAt={value.sys.createdAt} fields={value.fields} />)
              })}
            </List>
          </div>
        </div>
          : (page === 'SaveTab')
            ? <SaveTab getFilteredEntries={this.props.getFilteredEntries} user={this.props.user} />
            : (page === 'App')
              ? <App />
              : ''
        }
      </React.Fragment>
    )
  }
}

export default Tabs
