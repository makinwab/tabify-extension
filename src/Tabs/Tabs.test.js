import React from "react"
import { create } from "react-test-renderer"
import Tabs from "./Tabs"
import TabsList from "./TabsList"
import { shallow } from "enzyme"

it("renders <Tabs> without crashing", () => {
  const props = {
    getFilteredEntries: jest.fn(() => Promise.resolve({}))
  }
  expect(shallow(<Tabs {...props} />)).toMatchSnapshot()
})

it("renders <TabsList> within <Tabs>", () => {
  const props = {
    getFilteredEntries: jest.fn(() => Promise.resolve({}))
  }
  const component = create(<Tabs {...props} />)
  const rootInstance = component.root

  const childrenComponents = rootInstance.children
  expect(childrenComponents.length).not.toBe(0) //TODO: expect different behaviour
})

it("renders <TabsList> without crashing", () => {
  const props = {
    entry: {
      fields: {
        url: "http://abc.com",
        name: "tab title",
        tag: {
          fields: {
            name: "test"
          }
        }
      },
      sys: {
        id: 1
      }
    }
  }
  const TabsListComponent = create(<TabsList {...props} />)
  expect(TabsListComponent.toJSON()).toMatchSnapshot()
  expect(
    TabsListComponent.root.findAllByProps({ className: "item", id: 1 }).length
  ).toBe(1)
})

it("removes a tab when the appropriate button is clicked", () => {
  const props = {
    entry: {
      fields: {
        url: "http://abc.com",
        name: "tab title",
        tag: {
          fields: {
            name: "test"
          }
        }
      },
      sys: {
        id: 1
      }
    }
  }
  const component = create(<TabsList {...props} />)
  const rootInstance = component.root
  const button = rootInstance.findAllByType("button")
  const DOMEvent = {
    preventDefault: jest.fn(),
    currentTarget: {
      dataset: {
        entryId: 1
      }
    }
  }
  // console.log(rootInstance.findAllByProps({ className: "item" })[0].props)
  // console.log(rootInstance.findAllByProps({ className: "item" })[1].props)
  expect(button[1].props.className.includes("red")).toBe(true)
  button[1].props.onClick(DOMEvent)
  console.log(button[1].props)
  // console.log(rootInstance.findAllByProps({ className: "item", id: 1 }).length)
  // expect(component.children).toBe(1)
})
