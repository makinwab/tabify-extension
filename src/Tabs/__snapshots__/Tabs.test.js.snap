// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Tabs> without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    getFilteredEntries={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="Tabs"
        id="Tabs"
      >
        <div
          className="page-header"
        >
          <div
            className="menu-links"
          >
            <Label
              as="a"
              className="with-pointer"
              color="black"
              onClick={[Function]}
            >
              <Icon
                as="i"
                name="add"
              />
               Save Tab
            </Label>
          </div>
          <Icon
            as="i"
            className="with-pointer"
            name="home"
            onClick={[Function]}
            size="large"
          />
        </div>
        <div
          className="tabs-main"
        >
          <center
            className="search-box"
          >
            <Search
              icon="search"
              input="text"
              minCharacters={1}
              name="searchTerm"
              noResultsMessage="No results found."
              onSearchChange={[Function]}
              showNoResults={true}
              size="big"
              value=""
            />
          </center>
          <List
            relaxed={true}
          >
            <h3
              className="no-result"
            >
              Sorry! No Tabs Found
               
              <span
                aria-label="sad"
                role="img"
              >
                ðŸ˜¢
              </span>
            </h3>
          </List>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="page-header"
          >
            <div
              className="menu-links"
            >
              <Label
                as="a"
                className="with-pointer"
                color="black"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="add"
                />
                 Save Tab
              </Label>
            </div>
            <Icon
              as="i"
              className="with-pointer"
              name="home"
              onClick={[Function]}
              size="large"
            />
          </div>,
          <div
            className="tabs-main"
          >
            <center
              className="search-box"
            >
              <Search
                icon="search"
                input="text"
                minCharacters={1}
                name="searchTerm"
                noResultsMessage="No results found."
                onSearchChange={[Function]}
                showNoResults={true}
                size="big"
                value=""
              />
            </center>
            <List
              relaxed={true}
            >
              <h3
                className="no-result"
              >
                Sorry! No Tabs Found
                 
                <span
                  aria-label="sad"
                  role="img"
                >
                  ðŸ˜¢
                </span>
              </h3>
            </List>
          </div>,
        ],
        "className": "Tabs",
        "id": "Tabs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="menu-links"
              >
                <Label
                  as="a"
                  className="with-pointer"
                  color="black"
                  onClick={[Function]}
                >
                  <Icon
                    as="i"
                    name="add"
                  />
                   Save Tab
                </Label>
              </div>,
              <Icon
                as="i"
                className="with-pointer"
                name="home"
                onClick={[Function]}
                size="large"
              />,
            ],
            "className": "page-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Label
                  as="a"
                  className="with-pointer"
                  color="black"
                  onClick={[Function]}
                >
                  <Icon
                    as="i"
                    name="add"
                  />
                   Save Tab
                </Label>,
                "className": "menu-links",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "a",
                  "children": Array [
                    <Icon
                      as="i"
                      name="add"
                    />,
                    " Save Tab",
                  ],
                  "className": "with-pointer",
                  "color": "black",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "name": "add",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " Save Tab",
                ],
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "className": "with-pointer",
                "name": "home",
                "onClick": [Function],
                "size": "large",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <center
                className="search-box"
              >
                <Search
                  icon="search"
                  input="text"
                  minCharacters={1}
                  name="searchTerm"
                  noResultsMessage="No results found."
                  onSearchChange={[Function]}
                  showNoResults={true}
                  size="big"
                  value=""
                />
              </center>,
              <List
                relaxed={true}
              >
                <h3
                  className="no-result"
                >
                  Sorry! No Tabs Found
                   
                  <span
                    aria-label="sad"
                    role="img"
                  >
                    ðŸ˜¢
                  </span>
                </h3>
              </List>,
            ],
            "className": "tabs-main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Search
                  icon="search"
                  input="text"
                  minCharacters={1}
                  name="searchTerm"
                  noResultsMessage="No results found."
                  onSearchChange={[Function]}
                  showNoResults={true}
                  size="big"
                  value=""
                />,
                "className": "search-box",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "icon": "search",
                  "input": "text",
                  "minCharacters": 1,
                  "name": "searchTerm",
                  "noResultsMessage": "No results found.",
                  "onSearchChange": [Function],
                  "showNoResults": true,
                  "size": "big",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "center",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h3
                  className="no-result"
                >
                  Sorry! No Tabs Found
                   
                  <span
                    aria-label="sad"
                    role="img"
                  >
                    ðŸ˜¢
                  </span>
                </h3>,
                "relaxed": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Sorry! No Tabs Found",
                    " ",
                    <span
                      aria-label="sad"
                      role="img"
                    >
                      ðŸ˜¢
                    </span>,
                  ],
                  "className": "no-result",
                },
                "ref": null,
                "rendered": Array [
                  "Sorry! No Tabs Found",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-label": "sad",
                      "children": "ðŸ˜¢",
                      "role": "img",
                    },
                    "ref": null,
                    "rendered": "ðŸ˜¢",
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="Tabs"
          id="Tabs"
        >
          <div
            className="page-header"
          >
            <div
              className="menu-links"
            >
              <Label
                as="a"
                className="with-pointer"
                color="black"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="add"
                />
                 Save Tab
              </Label>
            </div>
            <Icon
              as="i"
              className="with-pointer"
              name="home"
              onClick={[Function]}
              size="large"
            />
          </div>
          <div
            className="tabs-main"
          >
            <center
              className="search-box"
            >
              <Search
                icon="search"
                input="text"
                minCharacters={1}
                name="searchTerm"
                noResultsMessage="No results found."
                onSearchChange={[Function]}
                showNoResults={true}
                size="big"
                value=""
              />
            </center>
            <List
              relaxed={true}
            >
              <h3
                className="no-result"
              >
                Sorry! No Tabs Found
                 
                <span
                  aria-label="sad"
                  role="img"
                >
                  ðŸ˜¢
                </span>
              </h3>
            </List>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="page-header"
            >
              <div
                className="menu-links"
              >
                <Label
                  as="a"
                  className="with-pointer"
                  color="black"
                  onClick={[Function]}
                >
                  <Icon
                    as="i"
                    name="add"
                  />
                   Save Tab
                </Label>
              </div>
              <Icon
                as="i"
                className="with-pointer"
                name="home"
                onClick={[Function]}
                size="large"
              />
            </div>,
            <div
              className="tabs-main"
            >
              <center
                className="search-box"
              >
                <Search
                  icon="search"
                  input="text"
                  minCharacters={1}
                  name="searchTerm"
                  noResultsMessage="No results found."
                  onSearchChange={[Function]}
                  showNoResults={true}
                  size="big"
                  value=""
                />
              </center>
              <List
                relaxed={true}
              >
                <h3
                  className="no-result"
                >
                  Sorry! No Tabs Found
                   
                  <span
                    aria-label="sad"
                    role="img"
                  >
                    ðŸ˜¢
                  </span>
                </h3>
              </List>
            </div>,
          ],
          "className": "Tabs",
          "id": "Tabs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="menu-links"
                >
                  <Label
                    as="a"
                    className="with-pointer"
                    color="black"
                    onClick={[Function]}
                  >
                    <Icon
                      as="i"
                      name="add"
                    />
                     Save Tab
                  </Label>
                </div>,
                <Icon
                  as="i"
                  className="with-pointer"
                  name="home"
                  onClick={[Function]}
                  size="large"
                />,
              ],
              "className": "page-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Label
                    as="a"
                    className="with-pointer"
                    color="black"
                    onClick={[Function]}
                  >
                    <Icon
                      as="i"
                      name="add"
                    />
                     Save Tab
                  </Label>,
                  "className": "menu-links",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "a",
                    "children": Array [
                      <Icon
                        as="i"
                        name="add"
                      />,
                      " Save Tab",
                    ],
                    "className": "with-pointer",
                    "color": "black",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "i",
                        "name": "add",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " Save Tab",
                  ],
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "className": "with-pointer",
                  "name": "home",
                  "onClick": [Function],
                  "size": "large",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <center
                  className="search-box"
                >
                  <Search
                    icon="search"
                    input="text"
                    minCharacters={1}
                    name="searchTerm"
                    noResultsMessage="No results found."
                    onSearchChange={[Function]}
                    showNoResults={true}
                    size="big"
                    value=""
                  />
                </center>,
                <List
                  relaxed={true}
                >
                  <h3
                    className="no-result"
                  >
                    Sorry! No Tabs Found
                     
                    <span
                      aria-label="sad"
                      role="img"
                    >
                      ðŸ˜¢
                    </span>
                  </h3>
                </List>,
              ],
              "className": "tabs-main",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Search
                    icon="search"
                    input="text"
                    minCharacters={1}
                    name="searchTerm"
                    noResultsMessage="No results found."
                    onSearchChange={[Function]}
                    showNoResults={true}
                    size="big"
                    value=""
                  />,
                  "className": "search-box",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "icon": "search",
                    "input": "text",
                    "minCharacters": 1,
                    "name": "searchTerm",
                    "noResultsMessage": "No results found.",
                    "onSearchChange": [Function],
                    "showNoResults": true,
                    "size": "big",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "center",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h3
                    className="no-result"
                  >
                    Sorry! No Tabs Found
                     
                    <span
                      aria-label="sad"
                      role="img"
                    >
                      ðŸ˜¢
                    </span>
                  </h3>,
                  "relaxed": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Sorry! No Tabs Found",
                      " ",
                      <span
                        aria-label="sad"
                        role="img"
                      >
                        ðŸ˜¢
                      </span>,
                    ],
                    "className": "no-result",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Sorry! No Tabs Found",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-label": "sad",
                        "children": "ðŸ˜¢",
                        "role": "img",
                      },
                      "ref": null,
                      "rendered": "ðŸ˜¢",
                      "type": "span",
                    },
                  ],
                  "type": "h3",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`renders <TabsList> without crashing 1`] = `
Array [
  <div
    className="item"
    id={1}
    onClick={[Function]}
    role="listitem"
  >
    <i
      aria-hidden="true"
      className="linkify large icon middle aligned list-icon"
      onClick={[Function]}
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        <a
          href="http://abc.com"
          rel="noopener noreferrer"
          target="_blank"
        />
        <span
          className="tabs-tag"
        >
           
          ( #
          test
           )
        </span>
      </div>
      <div
        className="note description"
      />
      <div
        className="tab-meta description"
      >
        Saved at 
      </div>
      <br />
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <button
          className="ui teal tiny basic vertical animated button"
          onClick={[Function]}
        >
          <div
            className="hidden content"
          >
            Edit
          </div>
          <div
            className="visible content"
          >
            <i
              aria-hidden="true"
              className="edit icon"
              onClick={[Function]}
            />
          </div>
        </button>
        Â 
        <button
          className="ui red tiny basic vertical animated button"
          data-entry-id={1}
          id="test-button"
          onClick={[Function]}
        >
          <div
            className="hidden content"
          >
            Remove
          </div>
          <div
            className="visible content"
          >
            <i
              aria-hidden="true"
              className="remove icon"
              onClick={[Function]}
            />
          </div>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="ui divider"
  />,
]
`;
